library/src/main/scala/scala/scalajs/js/Array.scala

```diff
@@ -183,4 +183,10 @@ object Array extends Object {
 
   /** Returns true if the given value is an array. */
   def isArray(arg: Any): Boolean = native
+
+  /** Creates a new array from the given iterable */
+  def from[A](iterable: Iterable[A]): Array[A] = native
+
+  /** Creates a new array from the given iterator */
+  def from[A](iterator: Iterator[A]): Array[A] = native
```

Comment:
I think `Array.from` doesn't accept Iterator (only requires `next` property) itself.

Iterators defined in the ES spec can be passed to `Array.from`, since those iterators also conform [Iterable interface](https://www.ecma-international.org/ecma-262/6.0/#sec-iterable-interface) by [having `@@iterator` property on `%IteratorPrototype%` object](https://www.ecma-international.org/ecma-262/6.0/#sec-%iteratorprototype%-@@iterator). If I create an object which only conforms Iterator interface, passing it to `Array.from` won't work as expected:

![image](https://user-images.githubusercontent.com/705555/75315367-f1ee5d00-58a5-11ea-8067-6ba043dad6ec.png)

TypeScript declares `IterableIterator` interface for this kind of objects.

https://github.com/microsoft/TypeScript/blob/56b6d0d6663b71588dea2cb5ccdd88ba904b0795/lib/lib.es2015.iterable.d.ts#L54-L56

Reply:
Thanks.
Removed `def from(iterable)` and used type cast in https://github.com/scala-js/scala-js/pull/3793/commits/7636b5b2150272aeb8949c9b88fc78c519f31432